### PR TITLE
CORE-20131: Make Strategy CordaSerializable

### PR DESCRIPTION
**Pull request by josephzunigadaly**

### CODE REVIEW

**Review by filipesoliveira — APPROVED** [review]
The test `TokenSelectionTests` is green.

File: ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/token/selection/TokenClaimCriteria.java [hunk 1]
```diff
@@ -94,6 +95,7 @@ public TokenClaimCriteria(
      * @param targetAmount   The minimum value for the sum of {@link ClaimedToken#getAmount()} for the selected tokens.
      * @param strategy       The token selection strategy to use.
      */
+    @ConstructorForDeserialization
     public TokenClaimCriteria(
```
Comment (vkolomeyko) [inline ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/token/selection/TokenClaimCriteria.java:99]: Should we add a test to assert that `TokenClaimCriteria` can be serialized?
I am happy if we do it as a separate PR after this one is merged.

**Review by vkolomeyko — APPROVED** [review]

File: ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/token/selection/TokenClaimCriteria.java (hunk 1 above)
Comment (josephzunigadaly) [reply ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/token/selection/TokenClaimCriteria.java:99]: I'll raise a different PR for that one.
